### PR TITLE
Reorganize network policy related content

### PR DESCRIPTION
**Pull request by jboxman**
- https://issues.redhat.com/browse/OSDOCS-794

This is a prelude to updates specific to OVN-Kubernetes for 4.6. This content has been static for a year, so it applies to all prior supported releases.

Additional bug fixes and such forthcoming under separate cover once this merges.

Pages:

- [Configuring multitenant mode with network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/multitenant-network-policy.html)
- [Creating default network policies for a new project](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/default-network-policy.html)
- [Deleting a network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/deleting-network-policy.html)
- [Editing a network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/editing-network-policy.html)
- [Viewing a network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/viewing-network-policy.html)
- [Creating a network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/creating-network-policy.html)
- [About network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/about-network-policy.html)

### CODE REVIEW

**Comment by ahardin-rh** [comment]
Checked preview build and confirmed all links are functioning and content is rendering as expected. Everything LGTM!

**Comment by jboxman** [comment]
Hi @anuragthehatter,

I updated the multitenant docs (`modules/nw-networkpolicy-multitenant-isolation.adoc`) to include an additional policy rule, `allow-same-namespace`:

- [Configuring multitenant mode with network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/multitenant-network-policy.html)

Does this addition make sense?

Thanks!

**Comment by jboxman** [comment]
Hi @juanluisvaladas,

I updated the procedure for multitenancy by using network policy:

- [Configuring multitenant mode with network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/multitenant-network-policy.html)

Does this look reasonable?

Thanks!

**Comment by anuragthehatter** [comment]
> Hi @anuragthehatter,
> 
> I updated the multitenant docs (`modules/nw-networkpolicy-multitenant-isolation.adoc`) to include an additional policy rule, `allow-same-namespace`:
> 
> * [Configuring multitenant mode with network policy](https://osdocs-794--ocpdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/multitenant-network-policy.html)
> 
> Does this addition make sense?
> 
> Thanks!

@jboxman That looks good to me. Thanks

**Review by juanluisvaladas — CHANGES_REQUESTED** [review]
The ones where I say this is important really need to be changed because they are technically wrong.
The others are just suggestions

**Comment by jboxman** [comment]
@juanluisvaladas, glad I asked, this is invaluable. Thanks!

(I'm going to be revising the content after this PR to include `egress` for OVN-Kubernetes and also `ipBlock` which are both absent, with a target of 4.6 GA.)

File: modules/nw-networkpolicy-delete.adoc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Module included in the following assemblies:
 //
-// networking/configuring-networkpolicy.adoc
+// * networking/network_policy/deleting-network-policy.adoc
```
[the remark this answers is not in the document]
Comment (jboxman) [reply modules/nw-networkpolicy-delete.adoc:3]: @juanluisvaladas, this was probably my first work with OCP, ever, so I'm not sure where that came from. I'll fix it. Thanks!

File: modules/nw-networkpolicy-edit.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+[id="nw-networkpolicy-edit_{context}"]
+
+= Editing a NetworkPolicy object
+
+You can edit a NetworkPolicy object in a namespace.
+
+.Prerequisites
+
+* A cluster using the OpenShift SDN network plug-in with `mode: NetworkPolicy`
+set. This mode is the default for OpenShift SDN.
```
[the remark this answers is not in the document]
Comment (jboxman) [reply modules/nw-networkpolicy-edit.adoc:14]: @juanluisvaladas, thanks, I'll fix this throughout. I'll probably update it again to explicitly mention OVN-Kubernetes for the GA as well.

**Comment by juanluisvaladas** [comment]
> (I'm going to be revising the content after this PR to include egress for OVN-Kubernetes and also ipBlock which are both absent, with a target of 4.6 GA.)

We don't have ipblock? That was included in 4.3: https://github.com/openshift/sdn/blob/release-4.3/pkg/network/node/networkpolicy.go#L538-L546 so when you include it, this probably should be backported all the way back.

When you add it please make sure to mention that we don't support except in ipblock in openshift-sdn (true for all versions).

Also in the delete document, I know it's not part of this PR, but instead of: `$ oc delete networkpolicy -l name=<policy-name> ` it should be `$ oc delete networkpolicy <policy-name> `. You're not guaranteed to have the label name, maybe you are deleting more policies than you want or maybe you are not deleting anything.

**Comment by jboxman** [comment]
@juanluisvaladas, ipBlock didn't make it yet; I updated the `oc delete` and force-pushed. Thanks for catching that. I'd wanted to revisit that, but hadn't realized how dangerous it was.

File: modules/nw-networkpolicy-multitenant-isolation.adoc [hunk 3]
```diff
@@ -17,11 +17,12 @@ set. This mode is the default for OpenShift SDN.
 
 .Procedure
 
-. Create the following files containing NetworkPolicy object definitions:
-.. A file named `allow-from-openshift-ingress.yaml` containing the following:
+. Create the following NetworkPolicy objects:
```
[the remark this answers is not in the document]
Comment (jboxman) [reply modules/nw-networkpolicy-multitenant-isolation.adoc:19]: @juanluisvaladas, I'll tackle that in a subsequent PR.

**Pull request merged**